app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -413,8 +413,9 @@ protected void onResume() {
             addButton.setOnClickListener(v -> {
                 Intent intent = new Intent(getApplicationContext(), ScanActivity.class);
                 Bundle bundle = new Bundle();
-                if (selectedTab != 0) {
-                    bundle.putString(LoyaltyCardEditActivity.BUNDLE_ADDGROUP, groupsTabLayout.getTabAt(selectedTab).getText().toString());
+                Group group = (Group) groupsTabLayout.getTabAt(selectedTab).getTag();
+                if (group != null) {
+                    bundle.putInt(LoyaltyCardEditActivity.BUNDLE_ADDGROUP, group._id);
```

Comment:
`getTabAt` can also return null, the tests are unhappy about that.

```suggestion
                TabLayout.Tab tab = groupsTabLayout.getTabAt(selectedTab);
                if (tab != null) {
                    Group group = (Group) groupsTabLayout.getTabAt(selectedTab).getTag();
                    if (group != null) {
                        bundle.putInt(LoyaltyCardEditActivity.BUNDLE_ADDGROUP, group._id);
```